sklearn/metrics/_classification.py

```diff
@@ -506,6 +506,104 @@ def multilabel_confusion_matrix(y_true, y_pred, sample_weight=None,
     return np.array([tn, fp, fn, tp]).T.reshape(-1, 2, 2)
 
 
+def most_confused_classes(y_true, y_pred, labels=None, max_rows=None):
+    """Most confused classes.
+
+    In multiclass and binary classification, this function computes the classes
+    which have been confused the most often by the classifier.
+
+    Parameters
+    ----------
+
+    y_true: array, shape = [n_samples]
+        Ground truth (correct) target values.
+
+    y_pred : array, shape = [n_samples]
+        Estimated targets as returned by a classifier.
+
+    labels : array, shape = [n_classes], optional
+        List of labels to index the confusion matrix. This may be used to reorder
+        or select a subset of labels.
+        If none is given, those that appear at least once
+        in ``y_true`` or ``y_pred`` are used in sorted order.
+
+    max_rows : int, optional
+        Maximum number of pairs of classes to return. Corresponds to the
+        number of rows in most_confused.
+        If none is given, all rows are returned.
+
+    Returns
+    -------
+    most_confused : ndarray of shape (n_pairs, 3)
+        The most confused classes list, with each row containing:
+        [class_1, class_2, count]
```

Comment:
If the class name is a string, this array won't work

Reply:
Perhaps returning multiple arrays is acceptable

Reply:
The actual behavior is as follows:
If y_true and y_pred are arrays of string, the function will work properly, only translating 'class_1' and 'class_2' to numbers from 0 to n_classes - 1
Since most_confused_classes() relies on confusion_matrix(), the user can use the argument labels (array of strings) to specify the order of the classes.

If I understand correctly, you think that the function should return something like:
[['class_1_name', 'class_2_name'], count] for every row?